### ISSUE TITLE
Error creating build container while building the Dockerfile from React Native: https://github.com/react-native-community/docker-android/blob/master/Dockerfile

### ISSUE THREAD

**Issue by XVilka** [containers/buildah#2048]
**Description**

Error creating build container while building the Dockerfile from React Native: https://github.com/react-native-community/docker-android/blob/master/Dockerfile

**Steps to reproduce the issue:**
1. Download https://github.com/react-native-community/docker-android/blob/master/Dockerfile
2. Run `podman build -t reactnative .`
3. Get this error

**Describe the results you received:**
```
STEP 1: FROM openjdk:8
Getting image source signatures
Copying blob ada5d61ae65d done
Copying blob 67b8714e1225 done
Copying blob 64b12da521a3 done
Copying blob 844c33c7e6ea done
Copying blob f8427fdf4292 done
Copying blob f025bafc4ab8 done
Copying blob 2e38df533772 done
Error: error creating build container: The following failures happened while trying to pull image specified by "openjdk:8" based on search registries in /etc/containers/registries.conf:
* "localhost/openjdk:8": Error initializing source docker://localhost/openjdk:8: pinging docker registry returned: Get https://localhost/v2/: dial tcp [::1]:443: connect: connection refused
* "docker.io/library/openjdk:8": Error writing blob: error storing blob to file "/var/tmp/storage822132733/5": read tcp 10.25.232.52:58002->104.18.123.25:443: read: connection reset by peer
* "registry.fedoraproject.org/openjdk:8": Error initializing source docker://registry.fedoraproject.org/openjdk:8: Error reading manifest 8 in registry.fedoraproject.org/openjdk: manifest unknown: manifest unknown
* "registry.access.redhat.com/openjdk:8": Error initializing source docker://registry.access.redhat.com/openjdk:8: Error reading manifest 8 in registry.access.redhat.com/openjdk: name unknown: Repo not found
* "registry.centos.org/openjdk:8": Error initializing source docker://registry.centos.org/openjdk:8: Error reading manifest 8 in registry.centos.org/openjdk: manifest unknown: manifest unknown
* "quay.io/openjdk:8": Error initializing source docker://quay.io/openjdk:8: Error reading manifest 8 in quay.io/openjdk: error parsing HTTP 404 response body: invalid character '<' looking for beginning of value: "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2 Final//EN\">\n<title>404 Not Found</title>\n<h1>Not Found</h1>\n<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>\n"
```

**Describe the results you expected:**

Successfully build the image

**Output of `rpm -q buildah` or `apt list buildah`:**

```
dnf info buildah
上次元数据过期检查：2:54:07 前，执行于 Thu 26 Dec 2019 12:38:36 PM CST。
可安装的软件包
Name         : buildah
Version      : 1.11.4
发布         : 2.fc31
Architecture : x86_64
Size         : 8.4 M
源           : buildah-1.11.4-2.fc31.src.rpm
Repository   : updates
Summary      : A command line tool used for creating OCI Images
URL          : https://buildah.io
协议         : ASL 2.0
Description  : The buildah package provides a command line tool which can be used to
             : * create a working container from scratch
             : or
             : * create a working container from an image as a starting point
             : * mount/umount a working container's root file system for manipulation
             : * save container's root file system layer to create a new image
             : * delete a working container or an image
```

**Output of `buildah version`:**

```
Version:         1.11.4
Go Version:      go1.13.1
Image Spec:      1.0.1-dev
Runtime Spec:    1.0.1-dev
CNI Spec:        0.4.0
libcni Version:  
image Version:   5.0.0
Git Commit:      
Built:           Thu Jan  1 08:00:00 1970
OS/Arch:         linux/amd64
```

**Output of `podman version` if reporting a `podman build` issue:**

```
podman version                                                                                                                                                                                                               15:33:28 
Version:            1.6.2
RemoteAPI Version:  1
Go Version:         go1.13.1
OS/Arch:            linux/amd64
```

**Output of `cat /etc/*release`:**

```
Fedora release 31 (Thirty One)
```

**Output of `uname -a`:**

```
Linux localhost.localdomain 5.3.16-300.fc31.x86_64 #1 SMP Fri Dec 13 17:59:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# This file is is the configuration file for all tools
# that use the containers/storage library.
# See man 5 containers-storage.conf for more information
# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# Path to an helper program to use for mounting the file system instead of mounting it
# directly.
#mount_program = "/usr/bin/fuse-overlayfs"

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"

# mountopt specifies comma separated list of extra mount options
mountopt = "nodev,metacopy=on"

# Remap-UIDs/GIDs is the mapping from UIDs/GIDs as they should appear inside of
# a container, to UIDs/GIDs as they should appear outside of the container, and
# the length of the range of UIDs/GIDs.  Additional mapped sets can be listed
# and will be heeded by libraries, but there are limits to the number of
# mappings which the kernel will allow when you later attempt to run a
# container.
#
# remap-uids = 0:1668442479:65536
# remap-gids = 0:1668442479:65536

# Remap-User/Group is a name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid or /etc/subgid file.  Mappings are set up starting
# with an in-container ID of 0 and the a host-level ID taken from the lowest
# range that matches the specified name, and using the length of that range.
# Additional ranges are then assigned, using the ranges which specify the
# lowest host-level IDs first, to the lowest not-yet-mapped container-level ID,
# until all of the entries have been used for maps.
#
# remap-user = "storage"
# remap-group = "storage"

[storage.options.thinpool]
# Storage Options for thinpool

# autoextend_percent determines the amount by which pool needs to be
# grown. This is specified in terms of % of pool size. So a value of 20 means
# that when threshold is hit, pool will be grown by 20% of existing
# pool size.
# autoextend_percent = "20"

# autoextend_threshold determines the pool extension threshold in terms
# of percentage of pool size. For example, if threshold is 60, that means when
# pool is 60% full, threshold has been hit.
# autoextend_threshold = "80"

# basesize specifies the size to use when creating the base device, which
# limits the size of images and containers.
# basesize = "10G"

# blocksize specifies a custom blocksize to use for the thin pool.
# blocksize="64k"

# directlvm_device specifies a custom block storage device to use for the
# thin pool. Required if you setup devicemapper.
# directlvm_device = ""

# directlvm_device_force wipes device even if device already has a filesystem.
# directlvm_device_force = "True"

# fs specifies the filesystem type to use for the base device.
# fs="xfs"

# log_level sets the log level of devicemapper.
# 0: LogLevelSuppress 0 (Default)
# 2: LogLevelFatal
# 3: LogLevelErr
# 4: LogLevelWarn
# 5: LogLevelNotice
# 6: LogLevelInfo
# 7: LogLevelDebug
# log_level = "7"

# min_free_space specifies the min free space percent in a thin pool require for
# new device creation to succeed. Valid values are from 0% - 99%.
# Value 0% disables
# min_free_space = "10%"

# mkfsarg specifies extra mkfs arguments to be used when creating the base.
# device.
# mkfsarg = ""

# use_deferred_removal marks devicemapper block device for deferred removal.
# If the thinpool is in use when the driver attempts to remove it, the driver 
# tells the kernel to remove it as soon as possible. Note this does not free
# up the disk space, use deferred deletion to fully remove the thinpool.
# use_deferred_removal = "True"

# use_deferred_deletion marks thinpool device for deferred deletion.
# If the device is busy when the driver attempts to delete it, the driver
# will attempt to delete device every 30 seconds until successful.
# If the program using the driver exits, the driver will continue attempting
# to cleanup the next time the driver is used. Deferred deletion permanently
# deletes the device and all data stored in device will be lost.
# use_deferred_deletion = "True"

# xfs_nospace_max_retries specifies the maximum number of retries XFS should
# attempt to complete IO when ENOSPC (no space) error is returned by
# underlying storage device.
# xfs_nospace_max_retries = "0"

# If specified, use OSTree to deduplicate files with the overlay backend
ostree_repo = ""

# Set to skip a PRIVATE bind mount on the storage home directory.  Only supported by
# certain container storage drivers
skip_mount_home = "false"
```

**Comment by TomSweeneyRedHat** [comment]
@XVilka this smells like something wasn't quite right with network connections to the registries that you were trying to pull from.  My `/etc/containers/registries.conf` file has this entry:
```
[registries.search]
registries = ['docker.io', 'registry.fedoraproject.org', 'registry.access.redhat.com', 'registry.centos.org', 'quay.io']
```
and it hit successfully on docker.io.  Can you retry and see if things are working for you now?  If not, can you try to change the FROM statement in your Dockerfile to look like:

```
FROM docker.io/openjdk:8
```

**Comment by XVilka** [comment]
@TomSweeneyRedHat I retried multiple times and the error quite consistent between runs. But after I added `FROM docker.io/openjdk:8` it works out of the box now. If I reset podman state with removing all images, containers, etc, then remove `docker.io` prefix, and run build again, the same error appears back, so it is pretty consistent.

**Comment by TomSweeneyRedHat** [comment]
@XVilka thanks for the follow up note.  Something is going wonky then with the repo resolution on the system.  Can you share/attach the registry conf there, especially the `[registries.search]` section?  I also neglected to ask, are you running this as root or as a rootless user?

**Comment by XVilka** [comment]
@TomSweeneyRedHat I ran this as a rootless user. Sadly, I will not have access to this machine until January 7 (I am on vacation since today), so I can't share the registry configuration, but I didn't change that file, it should be the default one in Fedora 31 package.

**Comment by TomSweeneyRedHat** [comment]
@XVilka no worries and thanks for the info.  I too am on vacation/pto until Jan 6th and probably won't dive into this more fully until then anyway.  It's nice to get the info together though in case someone else has time to dive in before then.

**Comment by dlaukamp** [comment]
I've encountered the same error, also on Fedora 31, rootless user (makes no difference though). Adding "dockier.io/" to the FROM statement in the dockerfile also fixed the problem.
Like @XVilka, I'm using the default registries.conf file:

```
[registries.search]
registries = ['docker.io', 'registry.fedoraproject.org', 'registry.access.redhat.com', 'registry.centos.org', 'quay.io']
```

**Comment by rhatdan** [comment]
Well we are recommending that people not use shortnames, whenever possible.  Were you using the same Dockerfile?  
Did you get the same error on openjdk:8?

**Comment by rhatdan** [comment]
If after it happened could you attempt to just pull the image using buildah pull?

**Comment by dlaukamp** [comment]
No, I was not using the same dockerfile. I tried to follow the official docker tutorial from [here](https://docs.docker.com/get-started/part2/) (Dockerfile listed at the top) using podman instead of docker. And no, I wasn't able to use the image after encountering the error (didn't use the buildah command, though).

**Comment by rhatdan** [comment]
Please show the log of what happened?

**Comment by dlaukamp** [comment]
So I changed my dockerfile back to the original (removing the "docker.io" in the FROM part) and run "podman build ...." again. Today, however, the command executes without error. So I can't reproduce it, sadly.

The error I had yesterday was basically the same as the one from the start post (except for the image name). A few "Copying blob <hash> done" and then weird network errors/ blob writing errors on all repositories. That's how I found this issue in the first place.

So I can't really add anything useful here, sorry. I'll keep using podman/ buildah and see if it reoccurs. Nevertheless, thanks for your help.
